wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -430,6 +431,32 @@ public Command getDefaultCommand(Subsystem subsystem) {
     return m_subsystems.get(subsystem);
   }
 
+  private void cancel(Command command) {
+    if (m_deferScheduleAndCancel) {
+      m_toCancel.add(command);
+      return;
+    }
+    if (command == null) {
+      DriverStation.reportWarning("Tried to cancel a null command", true);
+      return;
+    }
+    if (!isScheduled(command)) {
+      return;
+    }
+    m_deferScheduleAndCancel = true;
+    command.end(true);
+    for (Consumer<Command> action : m_interruptActions) {
+      action.accept(command);
+    }
+    m_scheduledCommands.remove(command);
+    m_requirements.keySet().removeAll(command.getRequirements());
+    m_watchdog.addEpoch(command.getName() + ".end(true)");
+    m_deferScheduleAndCancel = false;
+    while (!m_toCancel.isEmpty() && !m_deferScheduleAndCancel) {
+      cancel(m_toCancel.poll());
+    }
```

Comment:
This is out of scope of this method. If this is here, it defies the whole point of extracting the method. Both queues should be polled from whatever called this method.

Reply:
Java and C++ should be made to behave consistently- Currently (8bcdff9) Java polls the queue from `cancel(Command...)` and `run()`, while C++ polls it inside the `Cancel(Command*)` overload. However, it is kind of tricky to enforce that the caller polls the queue, since C++'s `Cancel(Command*)` is public. The best solution is probably a private method with a different name that is like Java's `cancelCommand`, which all of the public `Cancel` overloads call (as many times as appropriate) and then poll the queue (once per call).